clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -182,8 +190,21 @@ private SecurityStore maybeCreateNewKeystore(Map<String, ?> configs) {
             return null;
     }
 
+    private SecurityStore maybeCreateNewTruststore(Map<String, ?> configs) {
+        boolean truststoreChanged = Objects.equals(configs.get(SslConfigs.SSL_TRUSTSTORE_TYPE_CONFIG), truststore.type) ||
+                Objects.equals(configs.get(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG), truststore.path) ||
+                Objects.equals(configs.get(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG), truststore.password);
+
```

Comment:
Can someone double check this statement? Seems missing `!` before `Objects.equals`. Same issue with method `maybeCreateNewKeystore`.  @rajinisivaram @hachikuji

Reply:
@allenxiang Oops, will fix the check. Thank you!

Reply:
@allenxiang Submitted a PR for fixing this: https://github.com/apache/kafka/pull/5029. Thanks.